DependencyInjection/Configuration.php

```diff
@@ -40,26 +43,91 @@ public function getConfigTreeBuilder() : TreeBuilder
         $organizeMigrationModes = $this->getOrganizeMigrationsModes();
 
         $rootNode
+            ->fixXmlConfig('migration', 'migrations')
+            ->fixXmlConfig('migrations_path', 'migrations_paths')
             ->children()
-                ->scalarNode('dir_name')->defaultValue('%kernel.root_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
-                ->scalarNode('namespace')->defaultValue('Application\Migrations')->cannotBeEmpty()->end()
-                ->scalarNode('table_name')->defaultValue('migration_versions')->cannotBeEmpty()->end()
-                ->scalarNode('column_name')->defaultValue('version')->end()
-                ->scalarNode('column_length')->defaultValue(14)->end()
-                ->scalarNode('executed_at_column_name')->defaultValue('executed_at')->end()
-                ->scalarNode('all_or_nothing')->defaultValue(false)->end()
                 ->scalarNode('name')->defaultValue('Application Migrations')->end()
-                ->scalarNode('custom_template')->defaultValue(null)->end()
-                ->scalarNode('check_database_platform')->defaultValue(true)->end()
-                ->scalarNode('organize_migrations')->defaultValue(false)
+
+                ->arrayNode('migrations_paths')
+                    ->info('A list of namespace/path pairs where to look for migrations.')
+                    ->isRequired()
+                    ->requiresAtLeastOneElement()
+                    ->useAttributeAsKey('namespace')
+                    ->prototype('scalar')->end()
+                 ->end()
+
+                ->arrayNode('services')
+                    ->info('A set of services to pass to the underlying doctrine/migrations library, allowing to change its behaviour.')
+                    ->useAttributeAsKey('service')
+                    ->defaultValue([])
+                    ->validate()
+                        ->ifTrue(static function ($v) {
+                            return count(array_filter(array_keys($v), static function (string $doctrineService) : bool {
+                                return strpos($doctrineService, 'Doctrine\Migrations\\') !==0;
```

Comment:
This will fail if users configure a service with a custom ID instead of a class name. I don't think you can validate this in the configuration, but need to do it in the extension where you can resolve the class name of the service.

Reply:
If you notice, I'm using array_keys, so what I'm checking is the dependency name on doctrine side, that is always a class or an interface in the doctrine namespace